spring-cloud-vault-config/src/main/java/org/springframework/cloud/vault/config/ClientAuthenticationFactory.java

```diff
@@ -155,6 +159,24 @@ private ClientAuthentication awsEc2Authentication(VaultProperties vaultPropertie
 				restOperations);
 	}
 
+	private ClientAuthentication awsIamAuthentication(VaultProperties vaultProperties) {
+
+		AWSCredentials awsCredentials = new DefaultAWSCredentialsProviderChain()
+												.getCredentials();
+
+		AwsIamAuthenticationOptions.AwsIamAuthenticationOptionsBuilder awsIamAuthenticationOptionsBuilder = AwsIamAuthenticationOptions.builder();
+
+		if (vaultProperties.getAwsIam() != null && vaultProperties.getAwsIam().getVaultRole() != null)
+			awsIamAuthenticationOptionsBuilder.role(vaultProperties.getAwsIam().getVaultRole());
+
+		AwsIamAuthenticationOptions options = awsIamAuthenticationOptionsBuilder
```

Comment:
Are credentials likely to change while the container/VM is running? If so, then it would make sense to register `DefaultAWSCredentialsProviderChain` with `builder.authenticationProvider(…)`.

Reply:
You are right the credentials can change on both EC2 and ECS as AWS automatically rotates the keys for the IAM role user.  I have updated the code to pass the `DefaultAWSCredentialsProviderChain` into the `AwsIamAuthenticationOptions`.